process/tags_test.go

```diff
@@ -0,0 +1,108 @@
+package process
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestTagSerde(t *testing.T) {
+	encoder := NewTagEncoder()
+
+	a := encoder.Encode([]string{"one", "two", "three"})
```

Comment:
I think the code you wrote will work, but we should add some tests for unicode

https://docs.datadoghq.com/tagging/